File: mobile/lib/features/vendor_catalog/domain/entities/vendor.dart
class Vendor {
  const Vendor({
    required this.id,
    required this.userId,
    required this.name,
    required this.category,
    required this.city,
    required this.description,
    required this.priceFrom,
    required this.status,
    required this.ratingAvg,
    required this.ratingCount,
    required this.photoIds,
  });

  final String id;
  final String userId;
  final String name;
  final String category;
  final String city;
  final String description;
  final int priceFrom;
  final String status;
  final double ratingAvg;
  final int ratingCount;
  final List<String> photoIds;
}

class VendorSearchResult {
  const VendorSearchResult({
    required this.items,
    required this.total,
    required this.page,
    required this.limit,
  });

  final List<Vendor> items;
  final int total;
  final int page;
  final int limit;
}

class VendorQuery {
  const VendorQuery({
    this.query,
    this.category,
    this.city,
    this.priceMin,
    this.priceMax,
    this.ratingMin,
    this.sort,
    this.page = 1,
    this.limit = 20,
  });

  final String? query;
  final String? category;
  final String? city;
  final int? priceMin;
  final int? priceMax;
  final double? ratingMin;
  final String? sort;
  final int page;
  final int limit;
}

File: backend/services/core/internal/domain/vendor.go
// Package domain holds the core service's pure business entities. They have no
// dependencies on any project package outside this one.
package domain

import (
	"strings"
	"time"
)

// VendorStatus is the moderation state of a vendor profile.
type VendorStatus string

const (
	VendorPending  VendorStatus = "pending"
	VendorApproved VendorStatus = "approved"
	VendorRejected VendorStatus = "rejected"
)

// Valid reports whether the status is a known value.
func (s VendorStatus) Valid() bool {
	switch s {
	case VendorPending, VendorApproved, VendorRejected:
		return true
	}
	return false
}

// Vendor is the business profile owned by a vendor-role user living in
// auth-svc. UserID is therefore a foreign UUID with no DB constraint.
type Vendor struct {
	ID          string       `json:"id"`
	UserID      string       `json:"userId"`
	Name        string       `json:"name"`
	Category    string       `json:"category"`
	City        string       `json:"city"`
	Description string       `json:"description"`
	PriceFrom   int64        `json:"priceFrom"`
	Status      VendorStatus `json:"status"`
	RatingAvg   float64      `json:"ratingAvg"`
	RatingCount int          `json:"ratingCount"`
	PhotoIDs    []string     `json:"photoIds"`
	CreatedAt   time.Time    `json:"createdAt"`
	UpdatedAt   time.Time    `json:"updatedAt"`
}

// IsPublic reports whether the vendor should be visible in the public catalog.
func (v *Vendor) IsPublic() bool { return v.Status == VendorApproved }

// VendorInput captures user-supplied fields for create / update operations.
type VendorInput struct {
	Name        string
	Category    string
	City        string
	Description string
	PriceFrom   int64
}

// Normalize trims whitespace on text fields.
func (in *VendorInput) Normalize() {
	in.Name = strings.TrimSpace(in.Name)
	in.Category = strings.TrimSpace(in.Category)
	in.City = strings.TrimSpace(in.City)
	in.Description = strings.TrimSpace(in.Description)
}
